File: tools/js/validate-dict.js
import fs from "fs";
import Ajv from "ajv";

const schema = JSON.parse(
  fs.readFileSync("llmd-dcs-dictionary.schema.json", "utf8")
);

const dict = JSON.parse(
  fs.readFileSync(process.argv[2], "utf8")
);

const ajv = new Ajv({
  allErrors: true,
  strict: true
});

const validate = ajv.compile(schema);
const valid = validate(dict);

if (valid) {
  console.log("✅ Dictionary is valid.");
} else {
  console.error("❌ Dictionary validation failed:");
  console.error(validate.errors);
  process.exit(1);
}

File: tools/py/validate_dict.py
import json
import sys
from jsonschema import Draft202012Validator

with open("llmd-dcs-dictionary.schema.json", "r", encoding="utf-8") as f:
    schema = json.load(f)

with open(sys.argv[1], "r", encoding="utf-8") as f:
    dictionary = json.load(f)

validator = Draft202012Validator(schema)
errors = sorted(validator.iter_errors(dictionary), key=lambda e: e.path)

if not errors:
    print("✅ Dictionary is valid.")
else:
    print("❌ Dictionary validation failed:")
    for error in errors:
        print(f"- {list(error.path)}: {error.message}")
    sys.exit(1)